scripts/build_docker.sh

```diff
@@ -95,14 +95,27 @@ ln -P Dockerfile "$temp_dir/"
 
 cd "$temp_dir"
 
-build_args=(
-	--platform "$arch"
-	--build-arg "CODER_VERSION=$version"
-	--tag "$image_tag"
-)
-
 log "--- Building Docker image for $arch ($image_tag)"
-docker buildx build "${build_args[@]}" . 1>&2
+
+# Pull the base image, copy the /etc/group and /etc/passwd files out of it, and
+# add the coder group and user. We have to do this in a separate step instead of
+# using the RUN directive in the Dockerfile because you can't use RUN if you're
+# building the image for a different architecture than the host.
+docker pull --platform "$arch" alpine:latest
+
+temp_container_id="$(docker create --platform "$arch" alpine:latest)"
+docker cp "$temp_container_id":/etc/group ./group
+docker cp "$temp_container_id":/etc/passwd ./passwd
+docker rm "$temp_container_id"
+
+echo "coder:x:1000:coder" >> ./group
```

Comment:
Should we just run the proper commands in the container here instead then copy the files? Seems like it could be simpler.

Reply:
You can't run commands in the container because it's a different architecture. All of the binaries like `sh`, `echo`, `useradd`, `groupadd` etc. are for whatever architecture the image is.

Reply:
We could just check in the /etc/passwd and /etc/group files to the repo, but then they won't update if alpine changes anything, so I opted for this instead.

Reply:
Oh I mean you can run them in this container instead of doing the echo

Reply:
e.g. `docker exec $temp_container_id adduser`

Reply:
That won't work since the container is for a different architecture. Docker lets you create containers for different architectures but they won't start because of exec format errors. I'm taking advantage of that here to copy the files out of the image for the correct arch.

Reply:
I 🤦🤦🤦🤦🤦🤦🤦🤦 so hard when I realized how wrong I was.